### PR TITLE
Migrate/vuex to pinia

### PR DESCRIPTION
**Pull request by GVodyanov**
Close https://github.com/nextcloud/calendar/issues/5686 as part of https://github.com/nextcloud/groupware/issues/64

Continuation of https://github.com/nextcloud/calendar/issues/5685

TODO:
- [x] Migrate setting store
- [x] Update settings store usage
- [x] Migrate principals store
- [x] Update principals store usage
- [x] Migrate importState store
- [x] Update importState store usage
- [x] Migrate importFiles store
- [x] Update importFiles store usage
- [x] Migrate fetchedTimeRanges store
- [x] Update fetchedTimeRanges store usage
- [x] Migrate davRestrictions store
- [x] Update davRestrictions store usage (not found..?)
- [x] Migrate contacts store
- [x] Update contacts store usage (not found..?)
- [x] Migrate calendars store
- [x] Update calendars store usage
- [x] Migrate calendarObjects store
- [x] Update calendarObjects store usage 
- [x] Migrate calendarObjectsInstance store
- [x] Update calendarObjectsInstance store usage 
- [x] Remove `mapGetters`, `mapState`, etc.
- [x] Remove direct $store references
- [x] Recreate unit tests for all the stores
- [x] Manual testing (in progress)

### CODE REVIEW

File: src/components/CalendarGrid.vue [hunk 1]
```diff
@@ -87,20 +90,21 @@ export default {
...
-			showWeekends: state => state.settings.showWeekends,
-			showWeekNumbers: state => state.settings.showWeekNumbers,
-			slotDuration: state => state.settings.slotDuration,
-			showTasks: state => state.settings.showTasks,
-			timezone: state => state.settings.timezone,
-			modificationCount: state => state.calendarObjects.modificationCount,
-		}),
+		...mapState(useSettingsStore, [
+			'eventLimit',
+			'skipPopover',
```
Comment (st3iny) [inline src/components/CalendarGrid.vue:100]:
```suggestion
```

Unused

File: src/components/CalendarGrid.vue [hunk 2]
```diff
@@ -87,20 +90,21 @@ export default {
...
+			'showWeekends',
+			'showWeekNumbers',
+			'slotDuration',
+			'showTasks',
+			'timezone',
```
Comment (st3iny) [inline src/components/CalendarGrid.vue:105]:
```suggestion
```

Unused

**Review by st3iny — CHANGES_REQUESTED** [review]
First batch

I tested most changes locally.

File: npm
[the remark this answers is not in the document]
Comment (st3iny) [reply npm:1]: Done

File: src/components/AppNavigation/EditCalendarModal.vue [hunk 3]
```diff
@@ -125,14 +126,17 @@ export default {
...
+			const id = this.calendarsStore.editCalendarModal?.calendarId
 			if (!id) {
 				return undefined
 			}
 
-			return this.$store.getters.getCalendarById(id)
+			const foundCalendar = this.calendarsStore.getCalendarById(id)
+			this.calendarColor = foundCalendar.color
+			this.calendarName = foundCalendar.displayName
+			return foundCalendar
```
[the remark this answers is not in the document]
Comment (st3iny) [reply src/components/AppNavigation/EditCalendarModal.vue]: Done

File: src/components/AppNavigation/Settings.vue [hunk 4]
```diff
@@ -164,13 +165,13 @@ export default {
 		ActionCheckbox,
 		ActionLink,
 		AppNavigationSettings,
-		Multiselect,
```
[the remark this answers is not in the document]
Comment (st3iny) [reply src/components/AppNavigation/Settings.vue]: Done

File: src/store/calendars.js [hunk 5]
```diff
@@ -49,1132 +50,858 @@ import {
...
+			return null
+		},
+
+		/**
+		 *
+		 * @param {object} state the store data
+		 * @return {function({Boolean}, {Boolean}, {Boolean}): {Object}[]}
+		 */
+		sortedCalendarFilteredByComponents: (state) => (vevent, vjournal, vtodo) => {
+			return this.sortedCalendars.filter((calendar) => {
```
[the remark this answers is not in the document]
Comment (st3iny) [reply src/store/calendars.js]: Done

File: src/mixins/EditorMixin.js [hunk 6]
```diff
@@ -402,15 +402,18 @@ export default {
...
+				name: getPrefixedRoute(this.$route.name, 'CalendarView'),
 				params,
 			})
-			this.$store.commit('resetCalendarObjectInstanceObjectIdAndRecurrenceId')
+
+			this.calendarObjectInstanceStore.isNew = false
+			this.calendarObjectInstanceStore.calendarObject = null
+			this.calendarObjectInstanceStore.calendarObjectInstance = null
+			this.calendarObjectInstanceStore.existingEvent.objectId = null
+			this.calendarObjectInstanceStore.existingEvent.recurrenceId = null
```
Comment (st3iny) [inline src/mixins/EditorMixin.js]: I don't like inlining this into a mixin/component. I'll push a commit to move it back to the store.

Inlining within a store is okay, IMO.

**Review by st3iny — DISMISSED** [review]
Second batch. One more left.

**Review by st3iny — COMMENTED** [review]
Third batch done. Now, only manual testing is left.

**Comment by GVodyanov** [comment]
I also did some clicking around and everything seems to be working well, thanks @st3iny!

**Comment by miaulalala** [comment]
Error in ListView when opening the sidebar for a readonly event:

```
[Vue warn]: Missing required prop: "name"

found in

---> <NcAppSidebar>
       <EditSidebar> at src/views/EditSidebar.vue
         <NcContent>
           <Calendar> at src/views/Calendar.vue
             <App> at src/App.vue
               <Root> vue.runtime.esm.js:4625
[Vue warn]: Error in mounted hook: "TypeError: this.$refs.header is undefined"

found in

---> <NcAppSidebar>
       <EditSidebar> at src/views/EditSidebar.vue
         <NcContent>
           <Calendar> at src/views/Calendar.vue
             <App> at src/App.vue
               <Root> vue.runtime.esm.js:4625
TypeError: this.$refs.header is undefined
    focus NcAppSidebar-CQWODfsy.mjs:681
    mounted NcAppSidebar-CQWODfsy.mjs:562
    VueJS 18
    init vue-router.esm.js:3005
    init vue-router.esm.js:3004
    updateRoute vue-router.esm.js:2414
    transitionTo vue-router.esm.js:2263
    confirmTransition vue-router.esm.js:2402
    step vue-router.esm.js:2084
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    routeEnterGuard vue-router.esm.js:2542
    beforeRouteEnter EditorMixin.js:686
    routeEnterGuard vue-router.esm.js:2535
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    runQueue vue-router.esm.js:2095
    confirmTransition vue-router.esm.js:2397
    step vue-router.esm.js:2084
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    resolveAsyncComponents vue-router.esm.js:2162
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    step vue-router.esm.js:2091
    step vue-router.esm.js:2091
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    created CalendarGrid.vue:235
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    __WEBPACK_DEFAULT_EXPORT__ windowTitleService.js:42
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    runQueue vue-router.esm.js:2095
    confirmTransition vue-router.esm.js:2392
    transitionTo vue-router.esm.js:2260
    push vue-router.esm.js:2606
    push vue-router.esm.js:3036
    push vue-router.esm.js:3035
    handleEventClick eventClick.js:82
    __WEBPACK_DEFAULT_EXPORT__ eventClick.js:37
    trigger internal-common.js:3680
    node_modules calendar-main.js:183014
    node_modules calendar-main.js:183880
    listenBySelector internal-common.js:270
    EventClicking index.js:1783
    node_modules calendar-main.js:183098
    node_modules calendar-main.js:183098
    node_modules calendar-main.js:194522
    setRef internal-common.js:2380
    node_modules calendar-main.js:186150
    Preact 22
vue.runtime.esm.js:3065
[Vue warn]: Missing required prop: "name"

found in

---> <NcAppSidebar>
       <EditSidebar> at src/views/EditSidebar.vue
         <NcContent>
           <Calendar> at src/views/Calendar.vue
             <App> at src/App.vue
               <Root> vue.runtime.esm.js:4625
[Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup().

found in

---> <NcDialog>
       <AttachmentsList> at src/components/Editor/Attachments/AttachmentsList.vue
         <NcAppSidebarTab>
           <NcAppSidebarTabs>
             <Fragment>
               <NcAppSidebar>
                 <EditSidebar> at src/views/EditSidebar.vue
                   <NcContent>
                     <Calendar> at src/views/Calendar.vue
                       <App> at src/App.vue
                         <Root> vue.runtime.esm.js:4625
[Vue warn]: Missing required prop: "showHeader"

found in

---> <InviteesList> at src/components/Editor/Invitees/InviteesList.vue
       <NcAppSidebarTab>
         <NcAppSidebarTabs>
           <Fragment>
             <NcAppSidebar>
               <EditSidebar> at src/views/EditSidebar.vue
                 <NcContent>
                   <Calendar> at src/views/Calendar.vue
                     <App> at src/App.vue
                       <Root> vue.runtime.esm.js:4625
```

Same for month view:

```
[Vue warn]: Missing required prop: "name"

found in

---> <NcAppSidebar>
       <EditSidebar> at src/views/EditSidebar.vue
         <NcContent>
           <Calendar> at src/views/Calendar.vue
             <App> at src/App.vue
               <Root> vue.runtime.esm.js:4625
[Vue warn]: Error in mounted hook: "TypeError: this.$refs.header is undefined"

found in

---> <NcAppSidebar>
       <EditSidebar> at src/views/EditSidebar.vue
         <NcContent>
           <Calendar> at src/views/Calendar.vue
             <App> at src/App.vue
               <Root> vue.runtime.esm.js:4625
TypeError: this.$refs.header is undefined
    focus NcAppSidebar-CQWODfsy.mjs:681
    mounted NcAppSidebar-CQWODfsy.mjs:562
    VueJS 18
    init vue-router.esm.js:3005
    init vue-router.esm.js:3004
    updateRoute vue-router.esm.js:2414
    transitionTo vue-router.esm.js:2263
    confirmTransition vue-router.esm.js:2402
    step vue-router.esm.js:2084
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    routeEnterGuard vue-router.esm.js:2542
    beforeRouteEnter EditorMixin.js:686
    routeEnterGuard vue-router.esm.js:2535
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    runQueue vue-router.esm.js:2095
    confirmTransition vue-router.esm.js:2397
    step vue-router.esm.js:2084
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    resolveAsyncComponents vue-router.esm.js:2162
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    step vue-router.esm.js:2091
    step vue-router.esm.js:2091
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    created CalendarGrid.vue:235
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    __WEBPACK_DEFAULT_EXPORT__ windowTitleService.js:42
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    runQueue vue-router.esm.js:2095
    confirmTransition vue-router.esm.js:2392
    transitionTo vue-router.esm.js:2260
    push vue-router.esm.js:2606
    push vue-router.esm.js:3036
    push vue-router.esm.js:3035
    handleEventClick eventClick.js:82
    __WEBPACK_DEFAULT_EXPORT__ eventClick.js:37
    trigger internal-common.js:3680
    node_modules calendar-main.js:183014
    node_modules calendar-main.js:183880
    listenBySelector internal-common.js:270
    EventClicking index.js:1783
    node_modules calendar-main.js:183098
    node_modules calendar-main.js:183098
    registerInteractiveComponent internal.js:761
    componentDidMount internal.js:747
    Preact 4
    node_modules calendar-main.js:183260
    flushSync internal-common.js:2229
    node_modules calendar-main.js:183259
    drained internal-common.js:156
    tryDrain internal-common.js:138
    resume internal-common.js:127
    batchRendering index.js:2070
    changeView internal-common.js:4805
    created CalendarGrid.vue:228
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    step vue-router.esm.js:2088
    iterator vue-router.esm.js:2384
    __WEBPACK_DEFAULT_EXPORT__ windowTitleService.js:42
    iterator vue-router.esm.js:2362
    step vue-router.esm.js:2087
    runQueue vue-router.esm.js:2095
vue.runtime.esm.js:3065
[Vue warn]: Missing required prop: "name"

found in

---> <NcAppSidebar>
       <EditSidebar> at src/views/EditSidebar.vue
         <NcContent>
           <Calendar> at src/views/Calendar.vue
             <App> at src/App.vue
               <Root> vue.runtime.esm.js:4625
[Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup().

found in

---> <NcDialog>
       <AttachmentsList> at src/components/Editor/Attachments/AttachmentsList.vue
         <NcAppSidebarTab>
           <NcAppSidebarTabs>
             <Fragment>
               <NcAppSidebar>
                 <EditSidebar> at src/views/EditSidebar.vue
                   <NcContent>
                     <Calendar> at src/views/Calendar.vue
                       <App> at src/App.vue
                         <Root> vue.runtime.esm.js:4625
[Vue warn]: Missing required prop: "showHeader"

found in

---> <InviteesList> at src/components/Editor/Invitees/InviteesList.vue
       <NcAppSidebarTab>
         <NcAppSidebarTabs>
           <Fragment>
             <NcAppSidebar>
               <EditSidebar> at src/views/EditSidebar.vue
                 <NcContent>
                   <Calendar> at src/views/Calendar.vue
                     <App> at src/App.vue
                       <Root> vue.runtime.esm.js:4625
```

**Comment by st3iny** [comment]
Nice finds. Those are regressions from the nc-vue v8 migration and calendar editor redesign. I'll create a new PR.

I also just pushed a commit to fix uploading attachments.

**Review by st3iny — APPROVED** [review]
Let's go!!! :rocket: :moon:

**Pull request merged**